conans/cli/command.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+
+from conans.cli.cli import SmartFormatter
+from conans.errors import ConanException
+
+
+def info_handler(func):
+    def decorator(*args, **kwargs):
+        ret = {"error": None, "data": None}
+        try:
+            ret["data"] = func(*args, **kwargs)
+        except Exception as exc:
+            ret["error"] = exc
+            raise
+        finally:
+            return ret
+
+    return decorator
+
+
+class ConanCommand(object):
+    def __init__(self, method, group=None, **kwargs):
```

Comment:
If there is a known (and limited) set of _formatters_ and we want to pass them as arguments (it is nice in the decorator), then let's do this and Python itself will do the check. Also, make the `group` compulsory, we will make it optional in the decorator.

```suggestion
    def __init__(self, method, group, cli=None, json=None):
```

Reply:
Maybe, the `cli` _formatter_ should be compulsory? Can you think about a command that doesn't provide a minimal `cli` output?

Reply:
I wouldn't like to have a signature like:
```python
def __init__(self, method, group, cli=None, json=None, html=None, dot=None)
```
Where some of them apply to only one or two commands. I am promoting the ``formatters``, because that seems to be the same level of abstraction in the definition, a ``ConanCommand`` has a method, a group and some formatters.
It is possible that some commands do not have cli output, for example, the ``conan lock create`` will create a file, but I don't see any return in the ``conan_api`` call, and thus, the ``cli`` formatter maybe shouldn't have any ``cli`` output format.